sqlite/lib/SQLiteService.js

```diff
@@ -15,6 +15,19 @@ class SQLiteService extends SQLService {
         dbc.function('session_context', key => dbc[$session][key])
         dbc.function('regexp', { deterministic: true }, (re, x) => (RegExp(re).test(x) ? 1 : 0))
         dbc.function('ISO', { deterministic: true }, d => d && new Date(d).toISOString())
+
+        // define date and time functions in js to allow for throwing errors
+        const isTime = /^\d{1,2}:\d{1,2}:\d{1,2}$/
+        const hasTimezone = /([+-]\d{1,2}:?\d{0,2}|Z)$/
+        const throwNaN = n => { if (Number.isNaN(n)) { throw new Error(`Value does not contain a date "${n}"`) } else { return n } }
```

Comment:
I'd expect  from the function name that this function always throws.
Does it make sense to extract the `isNaN` check?

Reply:
If it was not clear from the function name I would rename the function to `throwsIfNaN`.

Reply:
I'd prefer to rename it.